artist.py

```diff
@@ -0,0 +1,72 @@
+import pygame
+import os
+from common_constants import Vector
+
+# in charge of drawing Game objects
+class Artist:
+    def __init__(self, game):
+        self.game = game
+
+        # pygame setup
+        os.environ['SDL_VIDEO_WINDOW_POS'] = "%d,%d" % (0,0) # locate the window in the upper left
+        pygame.init()
+        pygame.display.set_caption(f'{type(game).__name__} Reduction')
+
+        # compute constants
+
+        # this is the approximate height of the menu bar at the top of iOS
+        BUFFER = 28
+
+        # how much display space we have to work with
+        SCREEN_DIM = Vector(pygame.display.Info().current_w - BUFFER, pygame.display.Info().current_h - BUFFER)
+
+        # how big could each block dimension be if we used the whole display
+        FITTED_BLOCK_DIM = SCREEN_DIM / game.grid.dim
+
+        MIN_BLOCK_DIM = 15 # arbitrary cutoff for ease of visualization
+
+        # blocks need to be square and not smaller than the min
+        self.BLOCK_DIM = max(MIN_BLOCK_DIM, min(FITTED_BLOCK_DIM()))
+
+        # limit the max window dimension to fit on the display, and ensure it multiplies the block dimension
+        self.WINDOW_DIM = Vector(int(round(min(SCREEN_DIM.x - (SCREEN_DIM.x % self.BLOCK_DIM), self.BLOCK_DIM * game.grid.dim.x))), int(round(min(SCREEN_DIM.y - (SCREEN_DIM.y % self.BLOCK_DIM), self.BLOCK_DIM * game.grid.dim.y))))
```

Comment:
This is really hard to parse when it's nested. Feel free to rename vars if inaccurate. Also you don't need the `int()` call because `round()` without a second param creates an `int`.
```suggestion
        window_width = min(SCREEN_DIM.x - (SCREEN_DIM.x % self.BLOCK_DIM), self.BLOCK_DIM * game.grid.dim.x)
        window_height = min(SCREEN_DIM.y - (SCREEN_DIM.y % self.BLOCK_DIM), self.BLOCK_DIM * game.grid.dim.y)
        self.WINDOW_DIM = Vector(round(window_width), round(window_height))
```